prometheus_metrics.go

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2024 Cloudflare, Inc. All rights reserved.
+// SPDX-License-Identifier: BSD-3-Clause
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+type PrometheusConfig struct {
+	Host       string
+	Port       string
+	ScrapePath string
+	MetricName string
+}
+
+type PrometheusMetrics struct {
+	startedAt time.Time
+	histogram prometheus.ObserverVec
+}
+
+func (p *PrometheusMetrics) Fire(result string) {
+	observer := p.histogram.With(prometheus.Labels{"method": "unknown", "status": "unknown", "result": result})
+	p.observe(observer)
+}
+
+func (p *PrometheusMetrics) ResponseStatus(method string, status int) {
+	observer := p.histogram.With(prometheus.Labels{"method": method, "status": fmt.Sprint(status), "result": "unknown"})
+	p.observe(observer)
+}
+
+func (p *PrometheusMetrics) observe(observer prometheus.Observer) {
+	elapsed := time.Now().Sub(p.startedAt)
+	observer.Observe(float64(elapsed.Milliseconds()))
+}
+
+type PrometheusMetricsFactory struct {
+	metricName string
+}
+
+func NewPrometheusMetricsFactory(config PrometheusConfig) (MetricsFactory, error) {
+	serveMux := http.NewServeMux()
+	serveMux.Handle(config.ScrapePath, promhttp.Handler())
+	server := http.Server{
+		Addr:    config.Host + ":" + config.Port,
```

Comment:
You want to use [`net.JoinHostPort`](https://pkg.go.dev/net#JoinHostPort)

```suggestion
		Addr:    net.JoinHostPort(config.Host, config.Port),
```

Reply:
Sure. Do you also want to fix this in [`statsd_metrics.go`](https://github.com/cloudflare/privacy-gateway-server-go/blob/1d8081f775055dd2f078e81da52fa517608f4a58/statsd_metrics.go#L37)?